src/lib/tzcode/strftime.c

```diff
@@ -300,7 +300,15 @@ _fmt(char *buf, ssize_t size, const char *format, const struct tnt_tm *t,
 						width = 9;
 					}
 				} else {
-					width = width >= 9 ? 9 : width;
+					/*
+					 ** If width is more then some value
+					 ** (most often it equals to 2^31 - 1)
+					 ** then overflow happens. Let it fail
+					 ** and assign to width maximum value
+					 ** it may have in order to be consistent
+					 ** with e.g '%50f'.
+					 */
+					width = width >= 9 || width < 0 ? 9 : width;
```

Comment:
Looks inconsistent:

```
$ tarantool -e "print(require('datetime').new{}:format('%64424509541f'))"
000000000
$ tarantool -e "print(require('datetime').new{}:format('%64424509441f'))"
0
```

Let's detect overflow by looking at `strtol` return code and/or errno, like we do here, for example:

https://github.com/tarantool/tarantool/blob/d31796079c477aa7454e6942997efa88550b69f8/src/lib/vclock/vclock.c#L100-L103

Reply:
> Looks inconsistent

That's really strange if it works like that on your machine. I get the same output:

```
┌─> Patched version
└─>  | ./tarantool -e "print(_TARANTOOL); print(require('datetime').new{}:format('%64424509541f'))"
2.11.0-entrypoint-969-g87f60dd42
000000000

┌─> Non-patched
└─>  | tarantool -e "print(_TARANTOOL); print(require('datetime').new{}:format('%64424509541f'))"
2.11.0-entrypoint-968-gd31796079
000000000
```

Here's the explanation:
```
; 64424509541 - 2^32 * 15
	101
```

So, the width equals to 101, old part of code works, replacing width with 9. Inconsistency is not possible, AFAICS. 

However, I won't insist. If we want to process errno, I suppose we shouldn't throw an error, but just replace `width` with 9. Am I right?

Reply:
I ran the *same* patched build, but with different inputs (note 541 vs 441):

```
%64424509541f
%64424509441f
```

I'd expect it to return `000000000` in both cases, but it returns `0` in the second case. That's what I meant by "inconsistent behavior".

> If we want to process errno, I suppose we shouldn't throw an error, but just replace `width` with 9. Am I right?

Yes.